pkg/rancher-desktop/config/__tests__/commandLineOptions.spec.ts

```diff
@@ -0,0 +1,390 @@
+    newPrefs.kubernetes.version = origPrefs.kubernetes.version;
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('one option over two args should change only one value', () => {
+    const newPrefs = updateFromCommandLine(prefs, lockedSettings, ['--kubernetes.version', '1.23.7']);
+
+    expect(newPrefs.kubernetes.version).toBe('1.23.7');
+    newPrefs.kubernetes.version = origPrefs.kubernetes.version;
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('boolean option to true should change only that value', () => {
+    const newPrefs = updateFromCommandLine(prefs, lockedSettings, ['--kubernetes.options.flannel=true']);
+
+    expect(origPrefs.kubernetes.options.flannel).toBeFalsy();
+    expect(newPrefs.kubernetes.options.flannel).toBeTruthy();
+    newPrefs.kubernetes.options.flannel = false;
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('boolean option set to implicit true should change only that value', () => {
+    const newPrefs = updateFromCommandLine(prefs, lockedSettings, ['--kubernetes.options.flannel']);
+
+    expect(origPrefs.kubernetes.options.flannel).toBeFalsy();
+    expect(newPrefs.kubernetes.options.flannel).toBeTruthy();
+    newPrefs.kubernetes.options.flannel = false;
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('boolean option set to false should change only that value', () => {
+    const newPrefs = updateFromCommandLine(prefs, lockedSettings, ['--kubernetes.options.traefik=false']);
+
+    expect(origPrefs.kubernetes.options.traefik).toBeTruthy();
+    expect(newPrefs.kubernetes.options.traefik).toBeFalsy();
+    newPrefs.kubernetes.options.traefik = true;
+    expect(newPrefs).toEqual(origPrefs);
+  });
+
+  test('changes specified options', () => {
+    const optionsByPlatform: Record<string, Array<string | [string, string]>> = {
+      win32: [
+        '--virtualMachine.hostResolver',
+      ],
+      '!win32': [
+        '--application.adminAccess',
+        ['--application.pathManagementStrategy', 'rcfiles'],
+        '--virtualMachine.memoryInGB',
+        '--virtualMachine.numberCPUs',
+      ],
+      darwin: [
+        '--experimental.virtualMachine.socketVMNet',
+      ],
+      '*': [
+        '--application.debug',
+        '--application.telemetry.enabled',
+        '--application.updater.enabled',
+        '--application.autoStart',
+        '--application.startInBackground',
+        '--application.hideNotificationIcon',
+        '--application.window.quitOnClose',
+        '--containerEngine.allowedImages.enabled',
+        ['--containerEngine.name', 'containerd'],
+        '--kubernetes.port',
+        '--kubernetes.enabled',
+        '--kubernetes.options.traefik',
+        '--kubernetes.options.flannel',
+        '--portForwarding.includeKubernetesServices',
+        '--images.showAll',
+        ['--images.namespace', 'mangos'],
+        '--diagnostics.showMuted',
+      ],
+    };
+
+    for (const platform in optionsByPlatform) {
```

Comment:
I'm going to write these remaining `test.each` comments in a separate issue because this is pre-existing code that doesn't have much to do with the locked-field work.

Written up in Issue #4099